### PR TITLE
Show the oasis in help without needing a hacky aliasof

### PR DESCRIPTION
**Pull request by stevecotton**
Help shows terrain types in the section for their default_base
    
Instead of the aliasing hack that the oasis terrain used to use, have the help topic generator reuse the editor_default_base as a hint that it should be included in that section.
    
The terrain_type class gets a new boolean method, and more documentation. There's a couple of refactors to use the new method in old code too.

### CODE REVIEW

**Comment by Jonathan-Kelly** [comment]
> Currently has a merge of #8783, but will be rebased on top of that later.

Oops, looks like I forgot to add `submerge=0.3` to that before it merged. Should I wait for this to merge before addressing that oversight or just immediately do a follow up PR to add in the submerge effect for oasis?

**Comment by CelticMinstrel** [comment]
…why is "updating it right now on your PR" not an option? Seems like the best one to me.

**Comment by stevecotton** [comment]
Doing an immediate follow-up would be good, this can wait.

**Comment by stevecotton** [comment]
image

Wondering if a conjunct list is a good idea (the `and` in `, and Shallow Water`), given that the best-of / worst of lists are still plain comma-separated ones:
image

**Review by soliton- — APPROVED** [review]

**Pull request merged**
